dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1027,6 +1025,43 @@ void DeltaMergeStore::mergeDeltaAll(const Context & context)
     }
 }
 
+std::optional<DM::RowKeyRange> DeltaMergeStore::mergeDeltaBySegment(const Context & context, const RowKeyValue & start_key)
+{
+    auto dm_context = newDMContext(context, context.getSettingsRef(), /*tracing_id*/ "mergeDeltaBySegment");
+    if (start_key.is_common_handle != dm_context->is_common_handle)
+    {
+        return std::nullopt;
```

Comment:
Simply throw an exception. It is unexpected status.

Reply:
The `start_key` comes from the RPC request. From the TiFlash perspective maybe better to think all RPC callers to be a bad guy in order to isolate component errors. Otherwise a mistake in TiDB may cause TiFlash to crash.

Reply:
Or may be better to leave this check in the ManualCompactManager? We could keep a property that all calls to `DeltaMergeStore` should be checked before calling.

Reply:
IMO, it is better to rise some serious thing to notify the `unexpected status`, as it implies bugs usually. How about returning errors to TiDB when there are invalid params?

Reply:
BTW throw exception here should not crash TiFlash.

Reply:
I have moved this check to the TiFlash service side. Please take a look again, thanks!

Also added a test in the DM to ensure that, when common handle is given for a int handle table, there will be an exception thrown.